.github/workflows/gpu_test.yml

```
@@ -0,0 +1,48 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: GPU Test V3
+
+on:
+  push:
+    branches: [ v3 ]
+  pull_request:
+    branches: [ v3 ]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: py=${{ matrix.python-version }}, np=${{ matrix.numpy-version }}, deps=${{ matrix.dependency-set }}
+
+    runs-on: gpu-runner
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12']
+        numpy-version: ['1.24', '1.26', '2.0']
+        dependency-set: ["minimal", "optional"]
```

what if we did something like this for now: 
```suggestion
        python-version: ['3.11']
        numpy-version: ['2.0']
        dependency-set: ["gpu"]
```

---

This would be ideal. There is currently a `gpu` dependency set in `pyproject.toml`, but I'm not sure how to get hatch to pick it up.